mesonbuild/dependencies/base.py

```diff
@@ -846,6 +849,609 @@ def extract_libtool_shlib(self, la_file):
+            if not mo_file_line:
+                raise DependencyException('Failed to parse CMake trace')
+
+            loc = mo_file_line.end()
+
+            file = mo_file_line.group(1)
+            line = mo_file_line.group(2)
+            func = mo_file_line.group(3)
+            args = mo_file_line.group(4).split(' ')
+            args = list(map(lambda x: x.strip(), args))
+            args = list(map(lambda x: reg_genexp.sub('', x), args)) # Remove generator expressions
+
+            yield CMakeTraceLine(file, line, func, args)
+
+    def _call_cmake_real(self, args, env):
+        cmd = self.cmakebin.get_command() + args
+        tmpdir = tempfile.gettempdir()
+        p, out, err = Popen_safe(cmd, env=env, cwd=tmpdir)
+        rc = p.returncode
+        call = ' '.join(cmd)
+        mlog.debug("Called `{}` in {} -> {}".format(call, tmpdir, rc))
+
+        return rc, out, err
+
+    def _call_cmake(self, args, env=None):
+        if env is None:
+            fenv = env
+            env = os.environ
+        else:
+            fenv = frozenset(env.items())
+        targs = tuple(args)
+
+        # First check if cached, if not call the real cmake function
+        cache = CMakeDependency.cmake_cache
+        if (self.cmakebin, targs, fenv) not in cache:
+            cache[(self.cmakebin, targs, fenv)] = self._call_cmake_real(args, env)
+        return cache[(self.cmakebin, targs, fenv)]
+
+    @staticmethod
+    def get_methods():
+        return [DependencyMethods.CMAKE]
+
+    def check_cmake(self):
+        evar = 'CMAKE'
+        if evar in os.environ:
+            cmakebin = os.environ[evar].strip()
+        else:
+            cmakebin = 'cmake'
+        cmakebin = ExternalProgram(cmakebin, silent=True)
+        if cmakebin.found():
+            try:
+                p, out = Popen_safe(cmakebin.get_command() + ['--version'], cwd=tempfile.gettempdir())[0:2]
+                if p.returncode != 0:
+                    mlog.warning('Found CMake {!r} but couldn\'t run it'
+                                 ''.format(' '.join(cmakebin.get_command())))
+                    # Set to False instead of None to signify that we've already
+                    # searched for it and not found it
+                    cmakebin = False
+            except (FileNotFoundError, PermissionError):
+                cmakebin = False
+
+            cmvers = re.sub(r'\s*cmake version\s*', '', out.split('\n')[0]).strip()
+            invalid_version = False
+            if not version_compare(cmvers, CMakeDependency.class_cmake_version):
+                invalid_version = True
+        else:
+            cmakebin = False
+        if not self.silent:
+            if cmakebin and invalid_version:
+                mlog.log('Found CMake:', mlog.red('NO'), '(version of', mlog.bold(cmakebin.get_path()),
+                         'is', mlog.bold(cmvers) ,'but version', mlog.bold(CMakeDependency.class_cmake_version),
```

Comment:
__[Flake8]__

[E231] missing whitespace after ','

Reply:
Oh, I missed that. Should be fixed now.